fempy/sim/LPi/SimLambda1520.C

```diff
@@ -0,0 +1,293 @@
+    std::vector<std::tuple<int, int, int, int>> pions;
+    std::vector<std::tuple<int, int, int, int>> lambdas;
+
+    int pdgLambda1520 = 3124;
+    double massLambda1520 = TDatabasePDG::Instance()->GetParticle(pdgLambda1520)->Mass();
+    // Lifetime of TDatabasePDG in s, in Pythia the unity [mm/c] are used, so a conversion factor of 3*10^11 is needed
+    double lifetimeLambda1520 = 3*pow(10, 11)*TDatabasePDG::Instance()->GetParticle(pdgLambda1520)->Lifetime(); 
+    
+    auto fDecay = new TF1("fDecay", "exp(-x/[0] )", 0., pow(10,-10)); // mm/c
+    fDecay->FixParameter(0, lifetimeLambda1520);
+    fDecay->SetNpx(1500); 
+  
+    // Read histograms from Xi(1530) to simulate acceptance and pT of the particle
+    TFile *kinemXi1530 = TFile::Open(kinemFilePath.data(), "r");
+    TH1F * hAcc1530 = (TH1F*)kinemXi1530->Get("hAcc");
+    TH1F * hPt1530 = (TH1F*)kinemXi1530->Get("hPt");
+
+    // Lambda 1520
+    Particle lambda1520Feature;
+    lambda1520Feature.id(pdgLambda1520);
+    lambda1520Feature.m(massLambda1520);
+
+    for (int iEvent = 0; iEvent < nEvents; ++iEvent) {
+
+        if(iEvent%5000 == 0)
+          std::cout << "Processing event " << iEvent << endl;
+        
+        // initialize particle properties of Lambda(1520)
+        double tauLambda1520 = fDecay->GetRandom();
+        double phiLambda1520 = gRandom->Rndm() * 2 * TMath::Pi();
+        double etaLambda1520 = hAcc1530->GetRandom();
+        double thetaLambda1520 = 2 * TMath::ATan( TMath::Exp(-etaLambda1520) );
+        double ptLambda1520 = hPt1530->GetRandom();
+        double pxLambda1520 = ptLambda1520 * TMath::Cos(phiLambda1520);
+        double pyLambda1520 = ptLambda1520 * TMath::Sin(phiLambda1520);
+        double pzLambda1520 = ptLambda1520 * TMath::Cos(thetaLambda1520) * (1 / TMath::Sin(thetaLambda1520));
+        double pLambda1520 = TMath::Sqrt(ptLambda1520 * ptLambda1520 + pzLambda1520 * pzLambda1520);
+        double ELambda1520 = TMath::Sqrt(massLambda1520 * massLambda1520 + pLambda1520 * pLambda1520);
+        
+        Particle lambda1520;
+        lambda1520.id(pdgLambda1520);
+        lambda1520.status(81);
+        lambda1520.m(massLambda1520);
+        lambda1520.xProd(0.);
+        lambda1520.yProd(0.);
+        lambda1520.zProd(0.);
+        lambda1520.tProd(0.);
+        lambda1520.e(ELambda1520);
+        lambda1520.px(pxLambda1520);
+        lambda1520.py(pyLambda1520);
+        lambda1520.pz(pzLambda1520);
+        lambda1520.tau(tauLambda1520);
+        
+        // remove all particles generated in the event and append the Lambda(1520)
+        pythia.event.reset();
+        pythia.event.append(lambda1520);
+        int idPartLambda = pythia.event[1].id();
+        pythia.particleData.mayDecay(idPartLambda, true);
+        
+        // force the decay of the Lambda
+        pythia.moreDecays();
+
+
+        for(int iPart=2; iPart<pythia.event.size(); iPart++) {
+            if(pythia.event[iPart].id()==3122) {
```

Comment:
Consider using `if (testFails) continue;`  in order to reduce the number of nested scopes